app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,238 @@
+
+/*
+ * This regular expression which validates the recipe name
+ * should only allow alpha numeric and should not allow special characters
+ * for e.g. recipe1 - will be allowed , recipe@ - will not be allowed
+ */
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
+export const noErrorState: IRecipeNameErrorData = {
+  isRecipeNameError: false,
+  recipeNameErrorMessage: '',
+};
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState({
+    recipeName: '',
+    description: '',
+  });
+  const [recipeNameErrorData, setRecipeNameErrorDataState] = useState(noErrorState);
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState(true);
+  const [apiParams, setApiParams] = useState({
+    getRecipeByNameParams: {
+      context: '',
+      recipeName: '',
+    },
+    createRecipeParams: {
+      context: '',
+    },
+  });
+
+  /*
+   * TODO: This static data has to be removed when we have actual API data,
+   * then directly we will get that data from store as directives
+   */
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  const { response: recipeByNameResponse, error: recipeByNameError } = useFetch(
+    MyDataPrepApi.getRecipeByName,
+    apiParams.getRecipeByNameParams
+  );
+
+  const { response: createRecipeResponse, error: createRecipeError } = useFetch(
+    MyDataPrepApi.createRecipe,
+    apiParams.createRecipeParams,
+    {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    }
+  );
+
+  // This useEffect is for handling getRecipeByName API call response
+  useEffect(() => {
+    if (recipeByNameResponse) {
+      updateRecipeNameErrorData(
+        {
+          isRecipeNameError: true,
+          recipeNameErrorMessage: T.translate(`${PREFIX}.sameNameErrorMessage`).toString(),
+        },
+        recipeFormData
+      );
+    }
+  }, [recipeByNameResponse]);
+
+  // This useEffect is for handling getRecipeByName API call error
+  useEffect(() => {
+    if (recipeByNameError) {
+      if (recipeByNameError.statusCode === 404) {
+        updateRecipeNameErrorData(noErrorState, recipeFormData);
+      } else {
+        setSnackbar({
+          open: true,
+          isSuccess: false,
+          message: (recipeByNameError.response as Record<string, string>).message,
+        });
+      }
+    }
+  }, [recipeByNameError]);
+
+  // This useEffect is for handling createRecipe API call response
+  useEffect(() => {
+    if (createRecipeResponse) {
+      setShowRecipeForm(false);
+      setSnackbar({
+        open: true,
+        isSuccess: true,
+        message: `${recipeSteps.length} ${T.translate(`${PREFIX}.recipeSaveSuccessMessage`)}`,
+      });
+    }
+  }, [createRecipeResponse]);
+
+  // This useEffect is for handling createRecipe API call error
+  useEffect(() => {
+    if (createRecipeError) {
+      setShowRecipeForm(false);
```

Comment:
plz check the UX here, do we need to close the form on error or just throw the error and allow for further update in the form